buildspec.yml

```diff
@@ -22,18 +24,29 @@ phases:
       - GITHUB_BRANCH=${CODEBUILD_WEBHOOK_HEAD_REF#refs/heads/*}
   pre_build:
     commands:
-      - echo "export of the image tag for build and push purposes"
-      - export IMAGE_TAG="${GITHUB_BRANCH}.${COMMIT_HASH}"
       - echo "creating a clean environment"
       - make clean setup
   build:
     commands:
-      - echo "we need to tag the image correctly, and this should depend on the branch it's from, and / or the branch"
-      - echo "it's merged into."
-      - docker build -t ${IMAGE_BASE_NAME}:${IMAGE_TAG} .
+      - echo Build started on $(date)
+      - export DOCKER_IMG_TAG_BASE=${IMAGE_BASE_NAME}:${GITHUB_BRANCH}
+      - export DOCKER_IMG_TAG_HASH=${DOCKER_IMG_TAG_BASE}.${COMMIT_HASH}
```

Comment:
This won't work with CodeBuild, the variable `COMMIT_HASH` and `GITHUB_BRANCH` must be initialized in the pre_build phase and not in install, at this time `CODEBUILD_` vars are not yet initialized. Moreover the `GITHUB_BRANCH` variable should be set using git command instead of `CODEBUILD_WEBHOOK_HEAD_REF` which is not set in case of Codebuild manual build, use 

```
      # Reading git branch (the utility in the deploy script is unable to read it automatically on CodeBuild)
      # see https://stackoverflow.com/questions/47657423/get-github-git-branch-for-aws-codebuild
      - export GITHUB_BRANCH="$(git symbolic-ref HEAD --short 2>/dev/null)"
      - |
        if [ "${GITHUB_BRANCH}" = "" ] ; then
          GITHUB_BRANCH="$(git branch -a --contains HEAD | sed -n 2p | awk '{ printf $1 }')";
          export GITHUB_BRANCH=${GITHUB_BRANCH#remotes/origin/};
        fi
```
instead (see service-qrcode)

Reply:
Comment still valid

Reply:
It was indeed not yet done. I adapted it.